sources/Inputs/src/Contingencies.cpp

```diff
@@ -0,0 +1,223 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  Contingencies.cpp
+ *
+ * @brief Contingencies implementation file
+ *
+ */
+
+#include "Contingencies.h"
+
+#include "Log.h"
+#include "Message.hpp"
+
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+namespace dfl {
+namespace inputs {
+
+Contingencies::Contingencies(const boost::filesystem::path& filepath) {
+  loadDefinitions(filepath);
+  init();
+}
+
+void Contingencies::loadDefinitions(const boost::filesystem::path& filepath)
+{
+  try {
+    boost::property_tree::ptree tree;
+    boost::property_tree::read_json(filepath.native(), tree);
+
+    /**
+     * The JSON format for contingency definitions is inherited from Powsybl:
+     *
+     * {
+     *   "version" : "1.0",
+     *   "name" : "list",
+     *   "contingencies" : [ {
+     *     "id" : "contingency1",
+     *     "elements" : [ {
+     *       "id" : "element11",
+     *       "type" : "BRANCH"
+     *     } ]
+     *   }, {
+     *     "id" : "contingency2",
+     *     "elements" : [ {
+     *        "id" : "element21",
+     *        "type" : "GENERATOR"
+     *     }, {
+     *        "id" : "element22"
+     *        "type" : "LOAD"
+     *     } ]
+     *   }
+     * }
+     */
+
+    LOG(debug) << MESS(ContingenciesReadingFrom, filepath) << LOG_ENDL;
+    for (const boost::property_tree::ptree::value_type& v : tree.get_child("contingencies")) {
+      const boost::property_tree::ptree& ptcontingency = v.second;
+      LOG(debug) << ptcontingency.get<std::string>("id") << LOG_ENDL;
+
+      ContingencyDefinition contingency(ptcontingency.get<std::string>("id"));
+      bool valid = true;
+      for (const boost::property_tree::ptree::value_type& pte : ptcontingency.get_child("elements")) {
+        const auto element_id = pte.second.get<std::string>("id");
+        const auto element_type = pte.second.get<std::string>("type");
+        LOG(debug) << "  " << element_id << " (" << element_type << ")" << LOG_ENDL;
+
+        if (const auto has_type = elementTypeFromString(element_type)) {
```

Comment:
Done. 

Do not get the comment about direct use of `element.type`. In this piece of code `element.type` is only set if the string `strElementType` (changed the name from `element_type` to clarify the use of a given string value) can be interpreted as a known type in the corresponding enum.

Reply:
Sorry it wasn't clear. What I had in mind is to build ContingencyElementDefinition with in constructor the element.type to set it directly instead of putting it in another line

Reply:
Thanks for the clarification. The suggested simplification for building contingency elements has been applied.